packages/peregrine/lib/talons/AccountInformationPage/useAccountInformationPage.js

```diff
@@ -0,0 +1,122 @@
+
+    const [activeChangePassword, setActiveChangePassword] = useState(false);
+    const [updatedData, setUpdatedData] = useState(false);
+    const [isUpdateMode, setIsUpdateMode] = useState(false);
+
+    const { data: accountInformationData, error: loadDataError } = useQuery(
+        getCustomerInformationQuery,
+        { skip: !isSignedIn, fetchPolicy: 'cache-and-network' }
+    );
+
+    const [
+        setCustomerInformation,
+        {
+            data: customerInformationUpdated,
+            error: setCustomerInformationError,
+            loading: isSubmittingInfo
+        }
+    ] = useMutation(setCustomerInformationMutation);
+
+    const [
+        changeCustomerPassword,
+        { error: changeCustomerPasswordError, loading: isSubmittingPassword }
+    ] = useMutation(changeCustomerPasswordMutation);
+
+    const initialValues = useMemo(() => {
+        if (customerInformationUpdated) {
+            return {
+                customer: customerInformationUpdated.updateCustomer.customer
+            };
+        }
+        if (accountInformationData) {
+            return { customer: accountInformationData.customer };
+        }
+    }, [customerInformationUpdated, accountInformationData]);
+
+    const handleActivePassword = useCallback(() => {
+        setActiveChangePassword(true);
+    }, [setActiveChangePassword]);
+
+    const handleDeActivePassword = useCallback(() => {
+        setActiveChangePassword(false);
+    }, [setActiveChangePassword]);
+
+    const handleSubmit = useCallback(
+        async formValues => {
+            try {
+                await setCustomerInformation({
+                    variables: formValues
+                });
+
+                if (activeChangePassword) {
+                    await changeCustomerPassword({
+                        variables: {
+                            currentPassword: formValues.password,
+                            newPassword: formValues.newPassword
+                        }
+                    });
+                }
+                setUpdatedData(true);
+            } catch {
+                // we have an onError link that logs errors, and FormError already renders this error, so just return
+                // to avoid triggering the success callback
+                return;
+            }
+        },
+        [setCustomerInformation, changeCustomerPassword, activeChangePassword]
+    );
+
+    const handleCancelUpdate = useCallback(() => {
+        setIsUpdateMode(false);
+        handleDeActivePassword();
+    }, [setIsUpdateMode, handleDeActivePassword]);
+
+    const showUpdateMode = useCallback(() => {
+        setIsUpdateMode(true);
+    }, [setIsUpdateMode]);
+
+    useEffect(() => {
+        if (
+            updatedData &&
+            (!setCustomerInformationError ||
+                (activeChangePassword && !changeCustomerPasswordError))
+        ) {
+            handleCancelUpdate();
+            setUpdatedData(false);
+        }
+    }, [
```

Comment:
I'm a little confused about this effect. Can you explain it to me? Seems like we set `updatedData` to true when we submit new values like name or password. And then once it is true and we don't have an error, we call `cancelUpdate` which doesn't make sense since we only set `updatedData` to true once the update is complete.

Is this just to close the modal?

Reply:
I am using that state to check when the api complete and success to close dialog